src/hooks/useRegisterEmailForm.ts

```diff
@@ -0,0 +1,63 @@
+import {ChangeEvent, FormEvent, useState} from "react";
+import {useNavigate} from "react-router-dom";
+
+const createValidationInput = () => {
+    let newValidationInput = document.createElement('input');
+    newValidationInput.type = 'email';
+    newValidationInput.required = true;
+    return newValidationInput;
+}
+
+const useRegisterEmailForm = () => {
+    const [userEmail, setUserEmail] = useState<string>('');
+    let validationInput = createValidationInput();
+
+    const navigate = useNavigate();
+
+    const simpleEmailValidation = (email: string): boolean => {
+        return /\S+@\S+\.\S+/.test(email);
```

Comment:
I would suggest testing this regex. Something like this would work:
`Hell@W0r!d@my-cool/email.com`

<img width="1016" alt="Screenshot 2023-01-24 at 21 44 54" src="https://user-images.githubusercontent.com/4402762/214427845-cd11a357-2e56-45de-a3c7-7d8ee1768cb5.png">

Reply:
Hey Tai! Yes, that's true, very well caught, mate! 

And this format is actually on purpose (as discussed in the link on the function comment ([discussion](https://stackoverflow.com/a/13975255))).

There were many 'email address format' formal specifications over time (for example [RFC 2822](https://www.rfc-editor.org/rfc/rfc2822#section-3.4.1)), but this may change at any time (like RFC 5322, RFC 5321 and RFC 3696). The second mentioned one is matched by the following regex:

`(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])`

> (and it's already outdated).

I believe it's bad to do those tests manually, so I wouldn't like to 'hard code' this to a function .
That's why we are delegating the validation to the browser, by creating an input and testing the email on it (using the checkValidity function).

But if the browser is older than 2010 (in general), or 2015 for Edge (['checkValidity' browser support table](https://developer.mozilla.org/en-US/docs/Web/API/HTMLSelectElement/checkValidity)), it may not offer support to the function.

And in the end, the only real way to test if an email address exists is by sending an email to it, right? haha

So with all of this combined, I chose to just add a simple format validation, just so that it won't be a normal input field, but it's also not a huge regex that also may not be valid in the future.

I believe the system will rarely be used by un-updated browsers like that, but it's still good for us to choose the right validation. Do you have any suggestion on a regex structure for us to use @tainguyenbui ?